File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataModel.java
package com.mcreater.amclcore.api.mihoyo.genshin.model;

import com.google.gson.annotations.SerializedName;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class GenshinGachaDataModel {
    @SerializedName("all_avatar")
    private List<GenshinGachaDataAvatarModel> avatar;
    @SerializedName("all_weapon")
    private List<GenshinGachaDataWeaponModel> weapon;
    @SerializedName("all_set")
    private List<GenshinGachaDataModel> set;
    @SerializedName("weapon_level")
    private List<GenshinGachaDataLevelModel> weaponLevel;
    @SerializedName("set_level")
    private List<GenshinGachaDataLevelModel> setLevel;
    @SerializedName("avatar_element")
    private List<GenshinGachaDataLevelModel> avatarElement;
    @SerializedName("weapon_cat")
    private List<GenshinGachaDataLevelModel> weaponCat;
    @SerializedName("avatar_tag")
    private List<GenshinGachaDataLevelModel> avatarTag;


}

File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataModel.kt
package com.mcreater.amclcore.api.mihoyo.genshin.model

import com.google.gson.annotations.SerializedName

data class GenshinGachaDataModel(
    @SerializedName("all_avatar") var avatar: List<GenshinGachaDataAvatarModel?>? = null,
    @SerializedName("all_weapon") var weapon: List<GenshinGachaDataWeaponModel>? = null,
    @SerializedName("all_set") val set: List<GenshinGachaDataSetModel>? = null,
    @SerializedName("weapon_level") val weaponLevel: List<GenshinGachaDataLevelModel>? = null,
    @SerializedName("set_level") val setLevel: List<GenshinGachaDataLevelModel>? = null,
    @SerializedName("avatar_element") val avatarElement: List<GenshinGachaDataLevelModel>? = null,
    @SerializedName("weapon_cat") val weaponCat: List<GenshinGachaDataLevelModel>? = null,
    @SerializedName("avatar_tag") var avatarTag: MutableList<GenshinGachaDataLevelModel?>? = null
)
